src/SDL_hints.c

```diff
@@ -34,123 +31,153 @@ typedef struct SDL_HintWatch
 
 typedef struct SDL_Hint
 {
-    char *name;
     char *value;
     SDL_HintPriority priority;
     SDL_HintWatch *callbacks;
-    struct SDL_Hint *next;
 } SDL_Hint;
 
-static SDL_Hint *SDL_hints;
+static SDL_PropertiesID hint_props = 0;
 
-SDL_bool SDL_SetHintWithPriority(const char *name, const char *value, SDL_HintPriority priority)
+static SDL_PropertiesID GetHintProperties(void)
 {
-    const char *env;
-    SDL_Hint *hint;
-    SDL_HintWatch *entry;
+    // this isn't thread-safe, but I'm gambling we don't touch hints from two threads at once at startup.
+    if (!hint_props) {
+        hint_props = SDL_CreateProperties();
+    }
+    return hint_props;
+}
+
+static void SDLCALL CleanupHintProperty(void *userdata, void *value)
+{
+    SDL_Hint *hint = (SDL_Hint *) value;
+    SDL_free(hint->value);
+
+    SDL_HintWatch *entry = hint->callbacks;
+    while (entry) {
+        SDL_HintWatch *freeable = entry;
+        entry = entry->next;
+        SDL_free(freeable);
+    }
+    SDL_free(hint);
+}
 
+SDL_bool SDL_SetHintWithPriority(const char *name, const char *value, SDL_HintPriority priority)
```

Comment:
Shouldn't this be int, for consistency with the rest of the SDL API?

Reply:
Yes, this was driving me nuts. We inherited it from SDL2. I'll change that.

Reply:
This has been changed in e75bf28f1b029e4d125ebd490cd0f2bb972978b3 (in this PR, not on the main branch yet).